### PR TITLE
add alternative text descriptions for images in Testing

### LINKED ISSUE #3238: No alt-text on `/reproducible-research/testing`

### PR DESCRIPTION
**Pull request by tobyhodges**
### Summary

Fixes #3238 by adding alternative text descriptions for images in _Guide for Reproducible Research_ -> _Code Testing_

### List of changes proposed in this PR (pull-request)

- [x] add alt-text for images in `reproducible-research/testing.md`
- [x] add alt-text for images in `reproducible-research/testing/testing-exceptions.md`
    - alt-text descriptions in this file based on guidance provided in https://nightingaledvs.com/writing-alt-text-for-data-visualization/

### What should a reviewer concentrate their feedback on?

- [ ] Do the alt-text descriptions convey the purpose of the images in the context of their respective sections?
- [ ] Could they be improved, either by making them clearer or by adding more detail that would be important for someone who cannot view the image itself?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file.
- [x] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file: @tobyhodges

### CODE REVIEW

**Comment by tobyhodges** [comment]
I think this is good to go now. I had to fiddle around a bit to solve an error caused by those pesky `:`s in the headlines of the screengrabbed articles, so I won't be offended if you want to squash-merge this branch for a tidier project history.

**Comment by aleesteele** [comment]
Thank you so much for writing these. I'm going to do a proper review on this this week @tobyhodges 😄 

In the meantime, I'll add some breadcrumbs to #2630, which is an upcoming chapter about writing alt text for data visualisations, and #3144, flagging that we need some guidance in our guides about how to write alt text for scriberia images too ! ❤️

File: book/website/reproducible-research/testing.md [hunk 1]
```diff
@@ -25,14 +25,14 @@ Here's a couple of illustrations exemplifying of why should write tests:
...
-alt:
+alt: "Headline of a December 2006 news article by Greg Miller, published in Science, titled A Scientist's Nightmare: Software Problem Leads to Five Retractions"
 ---
 ```
 
 ```{figure}  ../figures/testing-motivation2.*
 ---
 name: testing-motivation2
-alt:
+alt: "News article  by Lisa Grossman, published on Wired.com in November 2010, describing an inconsistency between the units of force expected as output and input of two pieces of software that resulted in the loss of a weather satellite when it reached its destination at Mars. The piece is titled November 10, 1999: Metric Math Mistake Muffed Mars Meteorology Mission"
```
Comment (Arielle-Bennett) [inline book/website/reproducible-research/testing.md:35]:
```suggestion
alt: "News article by Lisa Grossman, published on Wired.com in November 2010, describing an inconsistency between the units of force expected as output and input of two pieces of software that resulted in the loss of a weather satellite when it reached its destination at Mars. The piece is titled November 10, 1999: Metric Math Mistake Muffed Mars Meteorology Mission"
```

**Review by Arielle-Bennett — APPROVED** [review]
Adjusted a double space but otherwise I think this is good to merge @tobyhodges :)

**Comment by aleesteele** [comment]
Thanks @tobyhodges - merging this now! 🚀

**Pull request merged**